tensorflow_addons/seq2seq/tests/attention_wrapper_test.py

```diff
@@ -297,683 +266,626 @@ def setUp(self):
+    decoder_inputs = np.random.randn(batch_size, decoder_max_time, input_depth).astype(
+        np.float32
+    )
+    encoder_outputs = np.random.randn(
+        batch_size, encoder_max_time, encoder_output_depth
+    ).astype(np.float32)
+
+    attention_mechanisms = []
+    for creator, depth in zip(create_attention_mechanisms, attention_mechanism_depths):
+        # Create a memory layer with deterministic initializer to avoid
+        # randomness in the test between graph and eager.
+        if create_query_layer:
+            create_attention_kwargs["query_layer"] = tf.keras.layers.Dense(
+                depth, kernel_initializer="ones", use_bias=False
             )
-        elif attention_layers is not None:
-            # Compute sum of attention_layers output depth.
-            attention_depth = sum(
-                attention_layer.compute_output_shape(
-                    [batch_size, cell_depth + encoder_output_depth]
-                )
-                .dims[-1]
-                .value
-                for attention_layer in attention_layers
+        if create_memory_layer:
+            create_attention_kwargs["memory_layer"] = tf.keras.layers.Dense(
+                depth, kernel_initializer="ones", use_bias=False
             )
-        else:
-            attention_depth = encoder_output_depth * len(create_attention_mechanisms)
-
-        decoder_inputs = np.random.randn(
-            batch_size, decoder_max_time, input_depth
-        ).astype(np.float32)
-        encoder_outputs = np.random.randn(
-            batch_size, encoder_max_time, encoder_output_depth
-        ).astype(np.float32)
 
-        attention_mechanisms = []
-        for creator, depth in zip(
-            create_attention_mechanisms, attention_mechanism_depths
-        ):
-            # Create a memory layer with deterministic initializer to avoid
-            # randomness in the test between graph and eager.
-            if create_query_layer:
-                create_attention_kwargs["query_layer"] = tf.keras.layers.Dense(
-                    depth, kernel_initializer="ones", use_bias=False
-                )
-            if create_memory_layer:
-                create_attention_kwargs["memory_layer"] = tf.keras.layers.Dense(
-                    depth, kernel_initializer="ones", use_bias=False
-                )
-
-            attention_mechanisms.append(
-                creator(
-                    units=depth,
-                    memory=encoder_outputs,
-                    memory_sequence_length=encoder_sequence_length,
-                    **create_attention_kwargs,
-                )
+        attention_mechanisms.append(
+            creator(
+                units=depth,
+                memory=encoder_outputs,
+                memory_sequence_length=encoder_sequence_length,
+                **create_attention_kwargs,
             )
+        )
 
-        with self.cached_session(use_gpu=True):
```

Comment:
Could we mark this as a TODO to enable `GPU` testing in #1713 for this

Reply:
We could, but I believe we don't need to test on gpu here. We don't depend on any custom op and we don't test in float16. I believe that this is tensorflow's job to do the right thing on gpu here, not ours. But I don't have a very strong opinion there, in the new setup, it's just going to be an additional decorator to add gpu testing.